doc/reference.rst

```diff
@@ -8,3 +8,20 @@ Sphinx-Gallery Reference
 
 .. autodata:: sphinxgallery.__version__
     :annotation: Sphinx Gallery current version
+
+.. include:: modules/generated/sphinxgallery.__version__.examples
+.. raw:: html
+
+        <div style='clear:both'></div>
```

Comment:
Out of interest, what is the worse that can happen if you always use the clearing trick, even when not needed ?

.example files are automatically generated so I don't really see a problem to add the clearing there. At least it ensures that just doing an include will just work and you don't have to remember to add the raw html additional content.